Gems/EMotionFX/Code/Source/Editor/Plugins/SimulatedObject/SimulatedObjectWidget.cpp

```diff
@@ -547,12 +550,92 @@ namespace EMotionFX
         }
 
         AZ_Assert(joint->GetSkeletonJointIndex() != InvalidIndex, "Expected skeletal joint index to be valid.");
-        const EMotionFX::Transform jointTransform = actorInstance->GetTransformData()->GetCurrentPose()->GetWorldSpaceTransform(joint->GetSkeletonJointIndex());
+        const EMotionFX::Transform jointTransform =
+            actorInstance->GetTransformData()->GetCurrentPose()->GetWorldSpaceTransform(joint->GetSkeletonJointIndex());
 
         DebugDraw& debugDraw = GetDebugDraw();
         DebugDraw::ActorInstanceData* drawData = debugDraw.GetActorInstanceData(actorInstance);
         drawData->Lock();
         drawData->DrawWireframeSphere(jointTransform.m_position, radius, color, jointTransform.m_rotation, 12, 12);
         drawData->Unlock();
     }
+
+    void SimulatedObjectWidget::Render(EMotionFX::ActorRenderFlagBitset renderFlags)
+    {
+        if (!m_actor || !m_actorInstance)
+        {
+            return;
+        }
+
+        const AZ::Render::RenderActorSettings& settings = EMotionFX::GetRenderActorSettings();
+        const bool renderSimulatedJoints = renderFlags[RENDER_SIMULATEJOINTS];
+        const AZStd::unordered_set<size_t>& selectedJointIndices = EMStudio::GetManager()->GetSelectedJointIndices();
+        if (renderSimulatedJoints && !selectedJointIndices.empty())
+        {
+            // Render the joint radius.
+            const size_t actorInstanceCount = GetActorManager().GetNumActorInstances();
+            for (size_t actorInstanceIndex = 0; actorInstanceIndex < actorInstanceCount; ++actorInstanceIndex)
+            {
+                ActorInstance* actorInstance = GetActorManager().GetActorInstance(actorInstanceIndex);
+                const Actor* actor = actorInstance->GetActor();
+                const SimulatedObjectSetup* setup = actor->GetSimulatedObjectSetup().get();
+                AZ_Assert(setup, "Expected a simulated object setup on the actor instance.");
+                const size_t objectCount = setup->GetNumSimulatedObjects();
```

Comment:
If this assert ever does hit, the next line will crash. Recommend an early out if setup is nullptr to prevent a total editor crash